From 78d0d122c47898c20a5209d0020674d28b42c513 Mon Sep 17 00:00:00 2001
From: Alex Williamson <alex.williamson@redhat.com>
Date: Fri, 10 Apr 2015 16:34:02 +0200
Subject: [PATCH 05/14] vfio: use correct runstate

Message-id: <20150410163402.15324.34238.stgit@gimli.home>
Patchwork-id: 64791
O-Subject: [RHEL7.2 qemu-kvm PATCH 5/8] vfio: use correct runstate
Bugzilla: 1210505
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

Upstream: ba29776fd8160a5c1c1892af5e237fc37aec3cf7

io-error is for block device errors; it should always be preceded
by a BLOCK_IO_ERROR event.  I think vfio wants to use
RUN_STATE_INTERNAL_ERROR instead.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/misc/vfio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/misc/vfio.c b/hw/misc/vfio.c
index 81e6594..f6e019c 100644
--- a/hw/misc/vfio.c
+++ b/hw/misc/vfio.c
@@ -3650,7 +3650,7 @@ static void vfio_err_notifier_handler(void *opaque)
                  __func__, vdev->host.domain, vdev->host.bus,
                  vdev->host.slot, vdev->host.function);
 
-    vm_stop(RUN_STATE_IO_ERROR);
+    vm_stop(RUN_STATE_INTERNAL_ERROR);
 }
 
 /*
-- 
1.8.3.1

